src/edge/recombine.py

```diff
@@ -512,7 +514,9 @@ def recombine_region(genome, region, min_homology_arm_length, op, need_new_fragm
 
     new_fragment_id = None
     with genome.update_fragment_by_fragment_id(region.fragment_id,
-                                               new_fragment=need_new_fragment) as f:
+                                               new_fragment=region.fragment_id not in is_new_fragment) as f:
```

Comment:
I would refactor this `region.fragment_id not in is_new_fragment` out above to actually be 
`is_new_fragment = region.fragment_id not in new_fragments_dict` and then pass `is_new_fragment`, this will make the code more readable and also shorten this with line to pass the flake8.